docusaurus-docs/community/policies/backwards-compatibility.md

```diff
@@ -1,9 +1,280 @@
+object with a leading underscore is not considered to be public. This convention
+is used in the Python community to designate an object as private.
+
+Private examples: - `_private_func_or_method` - `_PrivateClass` -
+`_PRIVATE_VAR`.
+
+Public examples: - `public_func_or_method` - `PublicClass` - `PUBLIC_VAR`.
+
+The highest-level entity determines the visibility level.
+
+For example:
+
+```py
+class _Private:
+  # everything is private here even without underscores
+  def this_is_also_private(self): pass
+```
+
+or
+
+```py
+  def _foo():
+    def inner():
+    # inner is also private - no way to call it without calling _foo, which is
+    # private.
+```
+
+Tests are never considered to be part of the public API. Any code within the
+`tests/` directory is always considered to be private.
+
+Developers are encouraged to make code private by default and only expose
+objects as public if there is an explicit need to do so. Keeping code private by
+default limits the public API that the conda-store project developers are
+committing to supporting.
+
+#### Deprecating Python APIs
+
+Under exceptional circumstances such as a serious security vulnerability which
+can't be fixed without breaking changes, it may be necessary to deprecate,
+remove, or introduce breaking changes to objects in the public Python API. 
+This should be avoided if possible.
+
+If the desire is to prevent a developer from relying on a part of the Python
+API, adding a warning to the documentation along with a recommended alternative
+and a comment in the code should be used rather than a deprecation or removal.
+
+```python 
+  """
+  This function is deprecated [reason/details], use [replacement] instead
+  """
+```
+If part of the Python API must be deprecated or removed, a deprecation warning
+should be added for at least one release before the endpoint is removed. This
+requirement may be waived in the case of a serious security vulnerability.
+
+The deprecation or removal should always be clearly communicated in release
+notes and documentation This should include:
+- version number of the release where this was deprecated
+- provide suggestions for alternatives (if possible).
+- provide a reason for the deprecation or removal (such
+  as a link to a CVE or issue that necessitated the removal
+
+#### Types of objects
+
+##### Modules
+
+A breaking change for a module means that any element of the module's public API
+has a breaking change.
+
+##### Classes
+
+In a public class, a breaking change is one that changes or removes attributes
+or methods or alters their meanings.  Rather than changing methods or
+attributes, new methods or attributes should be added if needed.  For example,
+if you wanted to change a user id from an `int`, to a `uuid`, a new attribute
+`User.uuid` should be added, and all new code should use `User.uuid`.  Existing
+methods can use the new attribute or methods as well as long as that doesn't
+introduce breaking changes for the method.
```

Comment:
```suggestion
In a public class, a breaking change is one that changes or removes attributes
or methods or alters their meanings. Rather than changing methods or
attributes, new methods or attributes should be added if needed. For example,
if you wanted to change a user id from an `int` to a `uuid`, a new attribute
`User.uuid` should be added, and all new code should use `User.uuid`. Existing
methods can use the new attribute or methods as well as long as that doesn't
introduce breaking changes for the method.
```